pkg/config/yaml.go

```diff
@@ -39,14 +38,14 @@ func GetYamlFromFile(cfgPath string, k0sVars constant.CfgVars) (clusterConfig *v
 }
 
 func ValidateYaml(cfgPath string, k0sVars constant.CfgVars) (clusterConfig *v1beta1.ClusterConfig, err error) {
-	if cfgPath != "" {
-		clusterConfig, err = v1beta1.FromYamlFile(cfgPath, k0sVars)
-	} else if isInputFromPipe() {
-		clusterConfig, err = v1beta1.FromYamlPipe(os.Stdin, k0sVars)
-	} else {
+	switch cfgPath {
+	case "-":
```

Comment:
I think this option deserves a small note in the flag definitions. Something like
```
flagset.StringVarP(&CfgFile, "config", "c", "", "config file (default: ./k0s.yaml). Use '-' to read the config from stdin")
```